src/shared/Git-Credential-Manager/Program.cs

```diff
@@ -11,6 +12,9 @@ public static class Program
     {
         public static void Main(string[] args)
         {
+            // Name the main thread of execution for TRACE2 tracing
+            Thread.CurrentThread.Name ??= "main";
```

Comment:
Not sure if .NET Framework support the syntax sugar for `??=`.

Also, does this mean if there's already a name we don't set it to `main`? Why not just always set it to `main`?

Reply:
Note that for helper UI apps we reserve the entry/primary thread for the UI dispatcher.

Reply:
It is important that the main thread (the one sending the "version", "start" and "exit" events be called "main" so that I can special case other items that are associated with it.

are UI helper apps spawned as child processes from the "main" thread?  that model works for me.  Git's hook processes work that way.  Really, all of Git creates children from the main thread -- threads are only really used for parallel algorithms IIRC.

Reply:
That is correct! Thanks for confirming. We will stick with this name, but I will remove the null check per @mjcheetham's comment.